others/mongodb-101.md

```diff
@@ -0,0 +1,233 @@
+
+Collections are like tables in SQL databases. Initialising a collection is also happening lazily like this:
+
+```> db.Student.insert({name: 'alex'})```
+
+now if we run again `show dbs` after we create a collection with a document we will see that the database appears in the list.
+ This is because at the moment that we inserted this document, the document was created, which in turn created the collection which in turn created the database.
+ 
+### Finding Documents
+ 
+Finding a document by a single attribute:
+ 
+```
+> db.Student.find({name: 'alex')
+```
+
+Querying embedded attributes:
+```
+> db.Student.find({"scores.midterm": {$gte: 40}} )
+```
+
+AND / OR queries:
+```
+> db.Student.find({"$or": [{"scores.midterm": {$gt: 60}}, {"scores.final": {$lte: 75}}]} )
+```
+
+In these two examples above we can see the operators GreaterThanEquals >= , GreaterThan > and LessThanEquals <= in action.
+
+All operators in MongoDB:
+https://docs.mongodb.com/manual/reference/operator/query/
+
+### Updating Documents
+
+Updating documents has 2 parts. The first one is finding the document(s) we want to update and the second one is modifying their values.
+
+(!) 
+```
+db.Student.update({name: 'alex'}, {name: 'alexander'})
+```
+
+What will this command do?
+
+REPLACE the first instance of document matching {name:'alex'} with the new {name:'alexander'} !!!!
+
+Definitely not what we want...
+
+```
+> db.Student.update({name: 'alex'}, {$set: {name: 'alexander'}})
+```
+
+We can also use other operators like $inc, $mul, $min, $max
+
+https://docs.mongodb.com/manual/reference/operator/update/#fields
+
+(!) Update by default will update only the first instance it matches.
+multi:true
+
+(!) We can get update to update or *create* the document by using upsert:true
+
+# Deleting documents
+
+Deleting a single or more documents is as simple as:
+```mongodb
+> db.Student.remove({"_id":ObjectId("5a99e1209056c9e237d071d9")})
+WriteResult({ "nRemoved" : 1 })
+```
+
+Deleting a whole collection:
+```db.Student.drop()```
+
+Deleting a database:
+```db.dropDatabase()``` to delete the current database
+
+
+# Lesson 4 - Administration
+
+## scripting the shell
+
+There are several ways to write scripts for MongoDB shell:
+- use `mongo < mongo_script.sh`
+- use `mongo mongo_script.js`
+
+What's the difference?
+
+The first one will pipe in our script and execute its lines one by one as if we were typing them in Mongo shell.
+
+The second one will evaluate our javascript and attempt to run it with Mongo shell.
+
+
+**Always** test scripts locally in MongoDB (and in general).
+
+**Always** test in staging before production.
+
+**Always** keep backups in production. And test that you can restore from these backups regualarly.
+
+All of these are important for every database system but even more important in MongoDB as...
+
+THERE IS NO ROLLBACK! There are no transactions. Once you delete something, it's gone.
+
+
+
+=======OPTIONAL==============
+
+## Aggregation
+
+SQL
```

Comment:
Is it possible to have a table comparing SQL queries with Mongo ones? also Concepts like these would be great to be accompanied by exercises